pkg/ccl/backupccl/backup_job.go

```diff
@@ -711,6 +715,7 @@ func (b *backupResumer) Resume(ctx context.Context, execCtx interface{}) error {
 			p.ExecCfg().DistSQLSrv.ExternalStorage,
 			details.EncryptionOptions,
 			statsCache,
+			details.CoordinatorLocation,
```

Comment:
won't this _always_ restrict locality to the coordinator's locality for the entire backup? Don't we only want that sometimes?

Reply:
CoordinatorLocation is badly named -- it is a locality filter that controls placement of the coordinator. I'm renaming it in parallel over on #99088